indexer/src/main/scala/scalding/BaseAlternateNamesJoinIntermediateJob.scala

```diff
@@ -0,0 +1,326 @@
+// Copyright 2014 Foursquare Labs Inc. All Rights Reserved.
+package com.foursquare.twofishes.scalding
+
+import com.twitter.scalding._
+import com.foursquare.twofishes._
+import org.apache.hadoop.io.LongWritable
+import com.twitter.scalding.typed.TypedSink
+import com.foursquare.hadoop.scalding.SpindleSequenceFileSource
+import com.foursquare.twofishes.util.Lists.Implicits._
+import com.foursquare.twofishes.util.NameNormalizer
+
+class BaseAlternateNamesJoinIntermediateJob(
+  name: String,
+  featureSources: Seq[String],
+  altNameSources: Seq[String],
+  args: Args
+) extends TwofishesIntermediateJob(name, args) {
+
+  val features = getJobOutputsAsTypedPipe[LongWritable, GeocodeServingFeature](featureSources).group
+  val altNames = getJobOutputsAsTypedPipe[LongWritable, FeatureNames](altNameSources).group
+
+  val countryInfoCachedFile = getCachedFileByRelativePath("downloaded/countryInfo.txt")
+  @transient lazy val countryNameMap = InMemoryLookupTableHelper.buildCountryNameMap(countryInfoCachedFile)
+  @transient lazy val countryLangMap = InMemoryLookupTableHelper.buildCountryLangsMap(countryInfoCachedFile)
+
+  val customRewritesCachedFile = getCachedFileByRelativePath("custom/rewrites.txt")
+  @transient lazy val nameRewritesMap = InMemoryLookupTableHelper.buildNameRewritesMap(Seq(customRewritesCachedFile))
+
+  val customShortensCachedFile = getCachedFileByRelativePath("custom/shortens.txt")
+  @transient lazy val nameShortensMap = InMemoryLookupTableHelper.buildNameShortensMap(Seq(customShortensCachedFile))
+
+  val customDeletesCachedFile = getCachedFileByRelativePath("custom/deletes.txt")
+  @transient lazy val nameDeletesList = InMemoryLookupTableHelper.buildNameDeletesList(Seq(customDeletesCachedFile))
+
+  def processFeatureName(
+    cc: String,
+    woeType: YahooWoeType,
+    lang: String,
+    name: String,
+    flags: Seq[FeatureNameFlags]
+  ): Seq[FeatureName] = {
+    if (lang != "post") {
+
+      val spaceRe = " +".r
+      def fixName(s: String) = spaceRe.replaceAllIn(s, " ").trim
+
+      // TODO(rahul): actually use flags
```

Comment:
Eventually. Even in the code this was copied from, there's been no need for this yet, though.